src/httpz.zig

```diff
@@ -929,11 +929,17 @@ test "websocket: upgrade" {
     try stream.writeAll(&websocket.frameBin("close"));
 
     var buf: [20]u8 = undefined;
-    const n = try stream.read(&buf);
-    try t.expectEqual(12, n);
-    try t.expectEqual(129, buf[0]);
-    try t.expectEqual(10, buf[1]);
-    try t.expectString("over 9000!", buf[2..12]);
+    {
+        const n = try stream.read(&buf);
+        try t.expectEqual(2, n);
+        try t.expectEqual(129, buf[0]);
+        try t.expectEqual(10, buf[1]);
+    }
+    {
+        const n = try stream.read(&buf);
+        try t.expectEqual(10, n);
+        try t.expectString("over 9000!", buf[0..n]);
+    }
```

Comment:
@karlseguin just wanted to follow up.  any idea why the 2nd stream.read() is necessary here?  without it this test doesn't pass.  does this new behavior seem reasonable?

i see that this test was added around 2 months ago when the websocket integration was done.  did this test pass then?

Reply:
Think I've had that test randomly break too. I don't see why I thought the initial read would only ever read 2 bytes...

I rewrote it:
https://github.com/karlseguin/http.zig/commit/f44d969a6b8059546348f95d4218b5c1e9d5ab56

Reply:
ok thanks for looking at it :+1: .  just wanted to follow up to make sure you noticed it since i don't have much experience with this project.

Reply:
I just upgraded to the latest zig, and this test is flaky again. I  _think_ it's related to [this change in Zig](https://github.com/ziglang/zig/commit/7680c5330cbc9141b9a5444e30c512b6068ab50d). Looking into it.

Reply:
https://github.com/karlseguin/http.zig/commit/1762121e5ff1791d5405c4484f5958b329eb9c39

Original code might have been wrong on OSX anyways